src/client/collection.rs

```diff
@@ -0,0 +1,357 @@
+        Ok(self
+            .with_collections_client(|mut collection_api| async move {
+                let result = collection_api.list_aliases(ListAliasesRequest {}).await?;
+                Ok(result.into_inner())
+            })
+            .await?)
+    }
+
+    pub async fn collection_cluster_info(
+        &self,
+        collection_name: impl ToString,
+    ) -> anyhow::Result<CollectionClusterInfoResponse> {
+        let collection_name = collection_name.to_string();
+        let collection_name_ref = collection_name.as_str();
+
+        Ok(self
+            .with_collections_client(|mut collection_api| async move {
+                let request = CollectionClusterInfoRequest {
+                    collection_name: collection_name_ref.to_string(),
+                };
+                let result = collection_api.collection_cluster_info(request).await?;
+                Ok(result.into_inner())
+            })
+            .await?)
+    }
+
+    pub async fn create_shard_key(
+        &self,
+        collection_name: impl AsRef<str>,
+        shard_key: &shard_key::Key,
+        shards_number: Option<u32>,
+        replication_factor: Option<u32>,
+        placement: &[u64],
+    ) -> anyhow::Result<CreateShardKeyResponse> {
+        let collection_name = collection_name.as_ref();
+
+        Ok(self
+            .with_collections_client(|mut collection_api| async move {
+                let result = collection_api
+                    .create_shard_key(CreateShardKeyRequest {
+                        collection_name: collection_name.to_string(),
+                        request: Some(CreateShardKey {
+                            shard_key: Some(ShardKey::from(shard_key.clone())),
+                            shards_number,
+                            replication_factor,
+                            placement: placement.to_vec(),
+                        }),
+                        timeout: None,
+                    })
+                    .await?;
+                Ok(result.into_inner())
+            })
+            .await?)
+    }
+
+    pub async fn delete_shard_key(
+        &self,
+        collection_name: impl AsRef<str>,
+        shard_key: &shard_key::Key,
+    ) -> anyhow::Result<DeleteShardKeyResponse> {
+        let collection_name = collection_name.as_ref();
+
+        Ok(self
+            .with_collections_client(|mut collection_api| async move {
+                let result = collection_api
+                    .delete_shard_key(DeleteShardKeyRequest {
+                        collection_name: collection_name.to_string(),
+                        request: Some(DeleteShardKey {
+                            shard_key: Some(ShardKey::from(shard_key.clone())),
+                        }),
+                        timeout: None,
+                    })
+                    .await?;
+                Ok(result.into_inner())
+            })
+            .await?)
+    }
```

Comment:
We might want to move this into a `sharding_keys` module, similar to this: https://github.com/qdrant/qdrant/blob/master/lib/collection/src/collection/sharding_keys.rs

Reply:
It can be a next step

Reply:
Yeah, we discussed to do this at the end to not mess up the following PRs that are already queued.